security_best_practices/security_best_practices.sp

```diff
@@ -0,0 +1,243 @@
+locals {
+  tailscale_common_tags = merge(local.tailscale_compliance__common_tags, {
+    tailscale = "true"
+  })
+}
+
+variable "api_key_id" {
+  description = "API Key ID for the tailnet key."
+  default     = ["fakeapikey1", "fakeapikey2"]
+}
+
+benchmark "security_best_practices" {
+  title         = "Tailscale Security Best Practices"
+  description   = "Tailscale has many security features you can use to increase your network security. This benchmark provides best practices for using these features to harden your Tailscale deployment."
+  documentation = file("./security_best_practices/docs/security_best_practices.md")
+  children = [
+    control.tailscale_acl_ssh_admin_roles_assigned,
+    control.tailscale_acl_ssh_check_mode_enabled,
+    control.tailscale_device_authorization_enabled,
+    control.tailscale_device_key_expire,
+    control.tailscale_device_network_boundary_protected,
+    control.tailscale_device_upgrade_clients_in_timely_manner,
+    control.tailscale_tailnet_acl_groups_used,
+    control.tailscale_tailnet_acl_tags_used,
+    control.tailscale_tailnet_key_unused,
+  ]
+
+  tags = merge(local.tailscale_common_tags, {
+    type = "Benchmark"
+  })
+}
+
+control "tailscale_acl_ssh_admin_roles_assigned" {
+  title       = "Assign admin roles"
+  description = "Assign user roles for managing Tailscale as appropriate, based on job function and for separation of duties. Tailscale provides multiple user roles that restrict who can modify your tailnet's configurations."
+  sql  = <<-EOT
+    with  tailscale_users as (
+      select
+        tailnet_name,
+        count(*) as tailnet_number
+      from
+        tailscale_acl_ssh
+      where
+        users ?| array['group:admin']and users ?| array['group:itadmin']and users ?| array['group:networkadmin']and users ?| array['group:auditor']
+      group by
+        tailnet_name
+    )
+    select
+      t.tailnet_name as resource,
+      case
+        when tu.tailnet_number > 0 then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when tu.tailnet_number > 0 then t.tailnet_name || ' has all admin roles assigned.'
+        else t.tailnet_name || ' does not have all admin roles assigned.'
+      end as reason
```

Comment:
For this query, and any others missing it, should we have the `tailnet_name` as an additional dimension?